# HG changeset patch
# User Rik <octave@nomad.inbox5.com>
# Date 1282341353 25200
#      Fri Aug 20 14:55:53 2010 -0700
# Node ID 860427ac9b7742ed9edc22b27acae53071bfbcac
# Parent  b64803a8be4e54f5885ed9843b4aad7b2d28070c
Add ability to pass -pthread through to linker in mkoctfile.

Bug #30605

diff --git a/ChangeLog b/ChangeLog
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,8 +1,12 @@
+2010-08-09  John Swensen <jpswensen@gmail.com>
+
+	* mkoctfile.in: Add ability to pass -pthread through to linker.
+
 2010-08-13  Jaroslav Hajek  <highegg@gmail.com>
 
 	* NEWS: Update.
 
 2010-08-09  Rik <octave@nomad.inbox5.com>
 
 	* examples/Makefile.am: Don't echo 'echo' command when building
 	octave.desktop.
diff --git a/mkoctfile.in b/mkoctfile.in
--- a/mkoctfile.in
+++ b/mkoctfile.in
@@ -279,16 +279,19 @@ EOF
       fi
     ;;
     -D*)
       defs="$defs $1"
     ;;
     -[lLR]* | -Wl,*)
       ldflags="$ldflags $1"
     ;;
+    -pthread)
+      ldflags="$ldflags $1"
+    ;;
     -M | -depend | --depend)
       depend=true
       compile=false
     ;;
     -o | -ouput | --output)
       shift
       if [ $# -gt 0 ]; then
         outputfile="$1"
